<?xml version="1.0"?>

<!--
 SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.

 SPDX-License-Identifier: Apache-2.0
-->

<!--
 Usage: Used for the nitros bridge image converter, 
        Listen to ROS 1 bridge message (Sent from ROS2) and convert to image message to be sent to ROS 1
        Listen to ROS 1 image topic and convert to ROS 1 bridge message to be sent to ROS 2
 Components:
  - ImageConverterNode
-->

<launch>
  <arg name="sub_image_name" default="/image" />
  <arg name="pub_image_name" default="/ros1_output_image" />
  <node pkg="nodelet"
        type="nodelet"
        name="standalone_nodelet" 
        args="manager"
        output="screen"/>
  <remap from="/ImageConverterNode/ros1_input_image" to="$(arg sub_image_name)"/>
  <remap from="/ImageConverterNode/ros1_output_image" to="$(arg pub_image_name)"/>
  <remap from="/ImageConverterNode/ros1_input_bridge_image" to="/ros1_input_bridge_image"/>
  <remap from="/ImageConverterNode/ros1_output_bridge_image" to="/ros1_output_bridge_image"/>
  <node pkg="nodelet"
        type="nodelet"
        name="ImageConverterNode"
        args="load isaac_ros_nitros_bridge_ros1/ImageConverterNode standalone_nodelet"
        output="screen"/>
</launch>